collect_app/src/main/java/org/odk/collect/android/activities/GeoTraceGoogleMapActivity.java

```diff
@@ -529,26 +529,20 @@ private void setupAutomaticMode() {
     }
 
     public void setGeoTraceMode(View view) {
-        boolean checked = ((RadioButton) view).isChecked();
-        int i = view.getId();
-        if (i == R.id.trace_manual) {
-            if (checked) {
+        if (((RadioButton) view).isChecked()) {
+            if (view.getId() == R.id.trace_manual) {
                 traceMode = 0;
                 timeUnits.setVisibility(View.GONE);
                 timeDelay.setVisibility(View.GONE);
                 timeDelay.invalidate();
                 timeUnits.invalidate();
-            }
-
-        } else if (i == R.id.trace_automatic) {
-            if (checked) {
+            } else if (view.getId() == R.id.trace_automatic) {
```

Comment:
One nice thing about a switch (even with only two cases) is you don’t repeat the `view.getId()` for each case, and you don’t need a local variable for the value. I’m not sure I like the “too few” rule, but I won’t object until I try to write one.

Reply:
`for each case` sounds too serious here since in the worst case it's for both (only two) cases (switches with three elements are allowed).